### ISSUE TITLE
Ability to swap diffthis panes

### ISSUE THREAD

**Issue by jamischarles** [lewis6991/gitsigns.nvim#585]
LOVING gitsigns. Seriously great work!

**Is your feature request related to a problem? Please describe.**
I use the `diffthis` view frequently. I always have to swap the panes because I expect the left pane to be the current buffer and the right pane to contain the base to compare against.

**Describe the solution you'd like**
I'd love if we could pass in a config value for swapping the pane order.

**Describe alternatives you've considered**
1. Autocmd:
```lua
local gs = require("gitsigns")
vim.api.nvim_create_user_command(
    'Diff',
    function()
      gs.diffthis('~') -- call same as :Gitsigns diffthis
      feedkeys("<C-W><C-x>", "m") -- Swap the panes
    end,
    { nargs = 0 }
)
```
I really thought this would work, but it doesn't swap the panes. I'm guessing it gets called too soon and I need to delay.

2. BufEnter and when it enters the gitsigns diff url, then run the "swap panes" command then.

Keep up the great work!

**Comment by lewis6991** [comment]
Yes, `diffthis` is asynchronous which means the call is non-blocking and doesn't finish until a few main loop iterations later.

One of the current goals is to add support for command modifiers to the Gitsigns command so we can do something like `:splitright Gitsigns diffthis`. At the same time, anything a command can do, the Lua function should be able to do too.

For now we can add an option to `diffthis`:

```
diffthis({base}, {opts})                                         *gitsigns.diffthis()*
```

Where setting something like `opts.splitright = true` will provide the behaviour you want.

**Comment by jamischarles** [comment]
@lewis6991 Thank you! 

I pulled down the latest and it works 🎉 !

Took me some trial error to realize I needed to set
`vim.o.splitright = true` in my lua config file.

**I did try variations like these:**
`gs.diffthis('~', {splitright = true, botright=true})`
and

```lua
require('gitsigns').setup {
  sign_priority = 20,
  diff_opts = {
    splitright = true,
    botright = true,
  }
```

Works great now! Thanks!

**Comment by lewis6991** [comment]
> 
> ```lua
> require('gitsigns').setup {
>   sign_priority = 20,
>   diff_opts = {
>     splitright = true,
>     botright = true,
>   }
> ```

Hmm, this shouldn't do anything. The only option pulled out of `diff_opts` here is `vertical`.

**Comment by lewis6991** [comment]
Ok, I worked on this a little and decided to revert the behaviour introduced in #586.

Now `diffthis` contains an `opts` argument:

```
diffthis({base}, {opts})                                 *gitsigns.diffthis()*
                Perform a |vimdiff| on the given file with {base} if it is
                given, or with the currently set base (index by default).

                If {base} is the index, then the opened buffer is editable and
                any written changes will update the index accordingly.

                Parameters: ~
                    {base}   (string|nil): Revision to diff against. Defaults
                             to index.
                    {opts}   (table|nil):
                             Additional options:
                             • {vertical}: {boolean}. Split window vertically. Defaults to
                             config.diff_opts.vertical. If running via command line, then
                             this is taken from the command modifiers.
                             • {split}: {string}. One of: 'aboveleft', 'belowright',
                             'botright', 'rightbelow', 'leftabove', 'topleft'. Defaults to
                             'aboveleft'. If running via command line, then this is taken
                             from the command modifiers.

```

Additionally if using via the `:Gitsigns` command, modifiers like `rightbelow` etc will now be respected.

E.g. `:rightbelow Gitsigns diffthis`

**Comment by lewis6991** [comment]
> ```lua
> require('gitsigns').setup {
>   diff_opts = {
>     split = "botright"
>   }
> }
> ```

This isn't supported, `split` can only be passed to `diffthis()`. See `:help gitsigns-config-diff_opts`

**Comment by jamischarles** [comment]
@lewis6991 Understood!
Awesome! 

I'll test `:botright Gitsigns diffthis` again when the commit is on main (didn't seem to work on `cmdarg` branch because a commit from `main` is missing)

**Comment by lewis6991** [comment]
Ah, I committed to the wrong branch. Should be on main now.

**Comment by jamischarles** [comment]
Aaaaalmost there lol. Thank you, hopefully this is helpful 😄 .

That removes the bug, though the swap doesn't happen for these 2 use cases:
`:botright Gitsigns diffthis`
`:Gitsigns diffthis ~ {'split': 'botright'}`

With either of these the split happens in the original layout.

**Comment by lewis6991** [comment]
Thanks for the feedback. I found another issue and raised #588.

`:Gitsigns diffthis ~ {'split': 'botright'}` isn't supported. It should be `:Gitsigns diffthis ~ split=botright`. Note I'm still working on the interface here so it isn't fully documented.

`:botright Gitsigns diffthis` and `:Gitsigns diffthis ~ split=botright` now both work for me.

**Comment by jamischarles** [comment]
@lewis6991 
Nice! I pulled down that branch.

**This now works** 🎉 
`:Gitsigns diffthis ~ split=botright`

**this doesn't** 
`:botright Gitsigns diffthis`

**Comment by lewis6991** [comment]
What version of neovim are you running?

**Comment by jamischarles** [comment]
0.7. Do I still need that `vim.opt.splitright = true` Setting?

**Comment by lewis6991** [comment]
Ok, you might need 0.8dev for that specific form to work.

You no longer need to set `splitright`

**Comment by jamischarles** [comment]
Ok. Sounds good. 

I'll test that out later when 0.8 is stable. I'll use one of the options that works for now.

Thank you!
